spec/client/ics-006-solo-machine-client/README.md

```diff
@@ -127,205 +147,178 @@ function latestClientHeight(clientState: ClientState): uint64 {
-  sequence: uint64,
-  data: bytes) {
-    path = applyPrefix(prefix, "ports/{portIdentifier}/channels/{channelIdentifier}/packets/{sequence}")
-    abortTransactionUnless(height == clientState.consensusState.sequence)
-    abortTransactionUnless(!clientState.frozen)
-    abortTransactionUnless(proof.timestamp >= clientState.consensusState.timestamp)
-    value = clientState.consensusState.sequence + clientState.consensusState.diversifier + proof.timestamp + path + data
-    assert(checkSignature(clientState.consensusState.pubKey, value, proof.sig))
-    clientState.consensusState.sequence++
-    clientState.consensusState.timestamp = proof.timestamp
-}
+`UpdateStateOnMisbehaviour` updates the function after receving valid misbehaviour:
 
-function verifyPacketAcknowledgement(
-  clientState: ClientState,
-  height: uint64,
-  prefix: CommitmentPrefix,
-  proof: CommitmentProof,
-  portIdentifier: Identifier,
-  channelIdentifier: Identifier,
-  sequence: uint64,
-  acknowledgement: bytes) {
-    path = applyPrefix(prefix, "ports/{portIdentifier}/channels/{channelIdentifier}/acknowledgements/{sequence}")
-    abortTransactionUnless(height == clientState.consensusState.sequence)
-    abortTransactionUnless(!clientState.frozen)
-    abortTransactionUnless(proof.timestamp >= clientState.consensusState.timestamp)
-    value = clientState.consensusState.sequence + clientState.consensusState.diversifier + proof.timestamp + path + acknowledgement
-    assert(checkSignature(clientState.consensusState.pubKey, value, proof.sig))
-    clientState.consensusState.sequence++
-    clientState.consensusState.timestamp = proof.timestamp
+```typescript
+function updateStateOnMisbehaviour(clientMessage: ClientMessage) {
+    // freeze the client
+    clientState.frozen = true
+    set("clients/{identifier}/clientState", clientState)
 }
+```
+
+### State verification functions
+
+All solo machine client state verification functions simply check a signature, which must be provided by the solo machine.
+
+Note that value concatenation should be implemented in a state-machine-specific escaped fashion.
 
-function verifyPacketReceiptAbsence(
-  clientState: ClientState,
+```typescript
+function verifyMembership(
+  // provided height is unnecessary for solomachine
+  // since clientState maintains the expected sequence
   height: uint64,
-  prefix: CommitmentPrefix,
+  // delayPeriod is unsupported on solomachines
+  // thus these fields are ignored
+  delayTimePeriod: uint64,
+  delayBlockPeriod: uint64,
   proof: CommitmentProof,
-  portIdentifier: Identifier,
-  channelIdentifier: Identifier,
-  sequence: uint64) {
-    path = applyPrefix(prefix, "ports/{portIdentifier}/channels/{channelIdentifier}/receipts/{sequence}")
-    abortTransactionUnless(height == clientState.consensusState.sequence)
+  path: CommitmentPath,
+  value: []byte): boolean {
+    // the expected sequence used in the signature
     abortTransactionUnless(!clientState.frozen)
```

Comment:
I believe in ibc-go our `VerifyMembership` and `VerifyNonMembership` assume the caller has checked that the client status is Active before calling this function. Should we no longer be assuming such a behaviour since the function could be called outside of the context of core IBC?

Reply:
No i think that's fine so long as its documented. Will leave this in here, because it is just a protective check